src/classes/RD2_ValidationService_TEST.cls

```diff
@@ -774,7 +774,155 @@ private with sharing class RD2_ValidationService_TEST {
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder()
+            .withContact(getContact().Id)
+            .withStatusClosed()
+            .build();
+    
+        String errMessage = '';
+        try {
+            Test.startTest();
+            insert rd;
+            Test.stopTest();
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        
+        String expectedMessage =  String.format(
+            System.Label.RD2_StatusMustBeMapped,
+            new String[]{rd.Status__c});
+
+        System.assert(errMessage.contains(expectedMessage),
+            'RD should fail to inserted when the status is not mapped:' + errMessage
+        );
+    }
+
+    /***
+    * @description Verifies that RD is updated successfully when Status is mapped
+    */
+    @isTest
+    private static void shouldUpdateRDWhenStatusIsMapped() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        RD2_StatusMapper_TEST.Stub mapperStub = new RD2_StatusMapper_TEST.Stub()
+            .withDefaultMapping();
+        RD2_ValidationService.statusMapper = (RD2_StatusMapper) Test.createStub(
+            RD2_StatusMapper.Class,
+            mapperStub
+        );
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder()
+            .withContact(getContact().Id)
+            .withStatusActive()
+            .build();
+        insert rd;
+        
+        
+        String errMessage = '';
+        try {
+            Test.startTest();
+            rd.Status__c = RD2_Constants.STATUS_CLOSED;
+            update rd;
+            Test.stopTest();
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+
+        System.assert(String.isBlank(errMessage),
+            'RD update should succeed when Status is mapped: ' + errMessage
+        );
+    }
+
+    /***
+    * @description Verifies that RD fail to update when Status is not mapped
+    */
+    @isTest
+    private static void shouldFailRDUpdateWhenStatusIsNotMapped() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        RD2_StatusMapper_TEST.Stub mapperStub = new RD2_StatusMapper_TEST.Stub()
+            .withStatusMapping(RD2_Constants.STATUS_ACTIVE, RD2_Constants.STATUS_ACTIVE);
+        RD2_ValidationService.statusMapper = (RD2_StatusMapper) Test.createStub(
+            RD2_StatusMapper.Class,
+            mapperStub
+        );
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder()
+            .withContact(getContact().Id)
+            .withStatusActive()
+            .build();
+        insert rd;
+        
+        
+        String errMessage = '';
+        try {
+            Test.startTest();
+            rd.Status__c = RD2_Constants.STATUS_LAPSED;
+            update rd;
+            Test.stopTest();
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+
+        String expectedMessage =  String.format(
```

Comment:
Let's create a helper method that will return error message so it can be used in this test and `shouldFailRDInsertWhenStatusIsNotMapped()`.